### PR TITLE
Add a timeout of 4 hours to release shared fixture lock

### PR DESCRIPTION
**Pull request by enrico-usai**
It happened multiple times (5+) that the tests were stuck for 9+ hours waiting for the fixture to be released.

With this patch we're releasing shared fixture locks after a timeout.

Tested with a fake timeout of 30secs:

```
2023-05-09 17:19:44,703 - ERROR - 5175 - test_multiple_nics[eu-west-1-c6in.metal-ubuntu1804-slurm-euw1-az3] - fixture_utils - Shared fixture vpc_stacks_shared has not been released in 4 hours, currently in use by 8 processes ({'gw7: 5632', 'gw1: 5233', 'gw5: 5518', 'gw2: 5298', 'gw3: 5380', 'gw4: 5438', 'gw0: 5175', 'gw6: 5571'}), destroying it.
2023-05-09 17:19:44,703 - INFO - 5175 - test_multiple_nics[eu-west-1-c6in.metal-ubuntu1804-slurm-euw1-az3] - fixture_utils - Terminating process 5632.
2023-05-09 17:19:44,704 - INFO - 5175 - test_multiple_nics[eu-west-1-c6in.metal-ubuntu1804-slurm-euw1-az3] - fixture_utils - Terminating process 5233.
2023-05-09 17:19:44,704 - INFO - 5175 - test_multiple_nics[eu-west-1-c6in.metal-ubuntu1804-slurm-euw1-az3] - fixture_utils - Terminating process 5518.
2023-05-09 17:19:44,705 - INFO - 5175 - test_multiple_nics[eu-west-1-c6in.metal-ubuntu1804-slurm-euw1-az3] - fixture_utils - Terminating process 5298.
2023-05-09 17:19:44,706 - INFO - 5175 - test_multiple_nics[eu-west-1-c6in.metal-ubuntu1804-slurm-euw1-az3] - fixture_utils - Terminating process 5380.
2023-05-09 17:19:44,706 - INFO - 5175 - test_multiple_nics[eu-west-1-c6in.metal-ubuntu1804-slurm-euw1-az3] - fixture_utils - Terminating process 5438.
2023-05-09 17:19:44,707 - INFO - 5175 - test_multiple_nics[eu-west-1-c6in.metal-ubuntu1804-slurm-euw1-az3] - fixture_utils - Terminating process 5175.
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/framework/fixture_utils.py [hunk 1]
```diff
@@ -122,6 +124,18 @@ def release(self):
...
+                logging.error(
+                    (
+                        "Shared fixture %s has not been released in 4 hours, "
+                        "currently in use by %d processes (%s), destroying it."
+                    ),
+                    self.name,
+                    data.counter,
+                    data.currently_using_processes,
+                )
+                break
```
Comment (francesco-giordano) [inline tests/integration-tests/framework/fixture_utils.py:148]: thought (non-blocking): what about to kill the "hanging" pid?

File: tests/integration-tests/framework/fixture_utils.py [hunk 2]
```diff
@@ -99,6 +99,7 @@ def release(self):
                 self._save_fixture_data(data)
                 return
 
+        timeout = time.time() + 4*30*30  # 4 hours from now
```
Comment (francesco-giordano) [inline tests/integration-tests/framework/fixture_utils.py]: Issue: `4*30*30` is 60 minutes. Am I missing something?

Comment (enrico-usai) [reply tests/integration-tests/framework/fixture_utils.py]: Sorry wrong cut and paste, fixed!

File: tests/integration-tests/framework/fixture_utils.py (hunk 1 above)
Comment (enrico-usai) [reply tests/integration-tests/framework/fixture_utils.py:148]: good call-out, addressed.

**Review by francesco-giordano — APPROVED** [review]

**Pull request merged**